pynicotine/gtkgui/search.py

```diff
@@ -482,18 +487,19 @@ def create_model(self):
             str,                  # (3)  h_speed
             str,                  # (4)  h_queue
             str,                  # (5)  directory
-            str,                  # (6)  filename
-            str,                  # (7)  h_size
-            str,                  # (8)  h_bitrate
-            str,                  # (9)  h_length
-            GObject.TYPE_UINT,    # (10) bitrate
-            str,                  # (11) fullpath
-            str,                  # (12) country
-            GObject.TYPE_UINT64,  # (13) size
-            GObject.TYPE_UINT,    # (14) speed
-            GObject.TYPE_UINT,    # (15) queue
-            GObject.TYPE_UINT,    # (16) length
-            str                   # (17) color
+            str,                  # (6)  file type icon
+            str,                  # (7)  filename
+            str,                  # (8)  h_size
+            str,                  # (9)  h_bitrate
+            str,                  # (10) h_length
+            GObject.TYPE_UINT,    # (11) bitrate
+            str,                  # (12) fullpath
+            str,                  # (13) country
+            GObject.TYPE_UINT64,  # (14) size
+            GObject.TYPE_UINT,    # (15) speed
+            GObject.TYPE_UINT,    # (16) queue
+            GObject.TYPE_UINT,    # (17) length
+            bool                  # (18) free slots
```

Comment:
These numeric column id's seem like they are a real pain to change, and makes a mess of the diff everytime we add something or want to make changes to the GUI.

Perhaps instead they can be mapped using a class or something, such as they could be altered in one place.

Reply:
This would be a nice change in the TreeView class, so we could always reference columns by the string column_ids instead of the numerical ones.